### ISSUE TITLE
Android build fails for version 4

### ISSUE THREAD

**Issue by jskrepnek** [intercom/intercom-cordova#206]
## Version info
  - Intercom for Cordova/PhoneGap version: 4.0.0
  - iOS/Android OS version: Android 7.1.2
  - iOS/Android Device: Nexus 5
  - Cordova Android platform version: 6.2.3

## Expected behavior
  - Builds

## Actual behavior
  - Doesn't build

## Steps to reproduce
 1. Upgrade plugin to version 4

## Logs

```
Build Date: 2017-09-18 22:00:41 +0000
--------------------------------------------------------------------------------
PLUGIN OUTPUT
--------------------------------------------------------------------------------
Fetching plugin "cordova-plugin-datepicker@0.9.2" via npm
Installing "cordova-plugin-datepicker" at "0.9.2" for android
Fetching plugin "cordova-plugin-whitelist@1.3.0" via npm
Installing "cordova-plugin-whitelist" at "1.3.0" for android

               This plugin is only applicable for versions of cordova-android greater than 4.0. If you have a previous platform version, you do *not* need this plugin since the whitelist will be built in.
Fetching plugin "cordova-plugin-splashscreen@4.0.3" via npm
Installing "cordova-plugin-splashscreen" at "4.0.3" for android
Fetching plugin "cordova-plugin-sslcertificatechecker@4.0.0" via npm
Installing "cordova-plugin-sslcertificatechecker" at "4.0.0" for android
Fetching plugin "cordova-plugin-appversion@1.0.0" via npm
Installing "cordova-plugin-appversion" at "1.0.0" for android
Fetching plugin "cordova-plugin-intercom@4.0.0" via npm
Installing "cordova-plugin-intercom" at "4.0.0" for android
Subproject Path: CordovaLib
Fetching plugin "cordova-plugin-statusbar@2.0.0" via npm
Installing "cordova-plugin-statusbar" at "2.0.0" for android
Fetching plugin "ionic-plugin-keyboard@2.2.1" via npm
Installing "ionic-plugin-keyboard" at "2.2.1" for android
Fetching plugin "cordova-plugin-geolocation@2.1.0" via npm
Installing "cordova-plugin-geolocation" at "2.1.0" for android
Fetching plugin "cordova-plugin-network-information@1.2.0" via npm
Installing "cordova-plugin-network-information" at "1.2.0" for android
Fetching plugin "cordova-plugin-inappbrowser@1.3.0" via npm
Installing "cordova-plugin-inappbrowser" at "1.3.0" for android
Fetching plugin "cordova-plugin-file-transfer@1.5.0" via npm
Installing "cordova-plugin-file-transfer" at "1.5.0" for android
Fetching plugin "cordova-plugin-file@^4.0.0" via npm
Installing "cordova-plugin-file" at "4.3.3" for android
Fetching plugin "cordova-plugin-compat@^1.0.0" via npm
Installing "cordova-plugin-compat" at "1.1.0" for android

The Android Persistent storage location now defaults to "Internal". Please check this plugin's README to see if your application needs any changes in its config.xml.

If this is a new application no changes are required.

If this is an update to an existing application that did not specify an "AndroidPersistentFileLocation" you may need to add:

      "<preference name="AndroidPersistentFileLocation" value="Compatibility" />"

to config.xml in order for the application to find previously stored files.
Plugin "cordova-plugin-file" already installed on android. Making it top-level.
Fetching plugin "cordova-plugin-file-opener2@2.0.2" via npm
Installing "cordova-plugin-file-opener2" at "2.0.2" for android
Fetching plugin "cordova-plugin-device@1.1.2" via npm
Installing "cordova-plugin-device" at "1.1.2" for android
Fetching plugin "urbanairship-cordova@6.6.0" via npm
Installing "urbanairship-cordova" at "6.6.0" for android
Subproject Path: CordovaLib
Fetching plugin "cordova-plugin-x-socialsharing@5.1.2" via npm
Installing "cordova-plugin-x-socialsharing" at "5.1.2" for android
Fetching plugin "cordova-universal-clipboard@0.1.0" via npm
Installing "cordova-universal-clipboard" at "0.1.0" for android
Fetching plugin "cordova-plugin-customurlscheme@4.2.0" via npm
Installing "cordova-plugin-customurlscheme" at "4.2.0" for android
Fetching plugin "cordova-plugin-touch-id@3.2.0" via npm
Installing "cordova-plugin-touch-id" at "3.2.0" for android
Fetching plugin "cordova-plugin-appsflyer-sdk@4.2.17" via npm
Installing "cordova-plugin-appsflyer-sdk" at "4.2.17" for android
Subproject Path: CordovaLib
Fetching plugin "https://github.com/kohofinancial/cordova-plugin-android-fingerprint-auth" via git clone
Using shallow clone
Repository "https://github.com/kohofinancial/cordova-plugin-android-fingerprint-auth" checked out to git ref "v1.2.8" at "aed2ec1".
Installing "cordova-plugin-android-fingerprint-auth" at "1.2.8" for android
--------------------------------------------------------------------------------
COMPILE OUTPUT
--------------------------------------------------------------------------------
Configuration on demand is an incubating feature.
:wrapper

BUILD SUCCESSFUL in 1s
1 actionable task: 1 executed
Subproject Path: CordovaLib
Configuration on demand is an incubating feature.
Incremental java compilation is an incubating feature.
Download https://jcenter.bintray.com/io/intercom/android/intercom-sdk-base/4.0.4/intercom-sdk-base-4.0.4.pom
Download https://maven.google.com/com/android/support/design/26.0.2/design-26.0.2.pom
Download https://maven.google.com/com/android/support/appcompat-v7/26.0.2/appcompat-v7-26.0.2.pom
Download https://maven.google.com/com/android/support/animated-vector-drawable/26.0.2/animated-vector-drawable-26.0.2.pom
Download https://maven.google.com/com/android/support/support-vector-drawable/26.0.2/support-vector-drawable-26.0.2.pom
Download https://maven.google.com/com/android/support/recyclerview-v7/26.0.2/recyclerview-v7-26.0.2.pom
Download https://jcenter.bintray.com/com/intercom/android-composer/1.3.1/android-composer-1.3.1.pom
Download https://jcenter.bintray.com/com/intercom/android-composer-gallery/1.3.1/android-composer-gallery-1.3.1.pom
Download https://maven.google.com/com/android/support/transition/26.0.2/transition-26.0.2.pom
Download https://maven.google.com/com/android/support/support-vector-drawable/26.0.2/support-vector-drawable-26.0.2.aar
Download https://maven.google.com/com/android/support/animated-vector-drawable/26.0.2/animated-vector-drawable-26.0.2.aar
Download https://maven.google.com/com/android/support/appcompat-v7/26.0.2/appcompat-v7-26.0.2.aar
Download https://maven.google.com/com/android/support/recyclerview-v7/26.0.2/recyclerview-v7-26.0.2.aar
Download https://maven.google.com/com/android/support/transition/26.0.2/transition-26.0.2.aar
Download https://maven.google.com/com/android/support/design/26.0.2/design-26.0.2.aar
Download https://jcenter.bintray.com/com/intercom/android-composer/1.3.1/android-composer-1.3.1.aar
Download https://jcenter.bintray.com/com/intercom/android-composer-gallery/1.3.1/android-composer-gallery-1.3.1.aar
Download https://jcenter.bintray.com/io/intercom/android/intercom-sdk-base/4.0.4/intercom-sdk-base-4.0.4.aar
:preBuild UP-TO-DATE
:preDebugBuild UP-TO-DATE
:checkDebugManifest
:preReleaseBuild UP-TO-DATE
:CordovaLib:preBuild UP-TO-DATE
:CordovaLib:preDebugBuild UP-TO-DATE
:CordovaLib:checkDebugManifest
:CordovaLib:prepareDebugDependencies
:CordovaLib:compileDebugAidl
:CordovaLib:compileDebugNdk UP-TO-DATE
:CordovaLib:compileLint
:CordovaLib:copyDebugLint UP-TO-DATE
:CordovaLib:mergeDebugShaders
:CordovaLib:compileDebugShaders
:CordovaLib:generateDebugAssets
:CordovaLib:mergeDebugAssets
:CordovaLib:mergeDebugProguardFiles
:CordovaLib:packageDebugRenderscript UP-TO-DATE
:CordovaLib:compileDebugRenderscript
:CordovaLib:generateDebugResValues
:CordovaLib:generateDebugResources
:CordovaLib:packageDebugResources
:CordovaLib:processDebugManifest
:CordovaLib:generateDebugBuildConfig
:CordovaLib:processDebugResources
:CordovaLib:generateDebugSources
:CordovaLib:incrementalDebugJavaCompilationSafeguard
:CordovaLib:compileDebugJavaWithJavac
:CordovaLib:compileDebugJavaWithJavac - is not incremental (e.g. outputs have changed, no previous execution, etc.).
Note: Some input files use or override a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
:CordovaLib:processDebugJavaRes UP-TO-DATE
:CordovaLib:transformResourcesWithMergeJavaResForDebug
:CordovaLib:transformClassesAndResourcesWithSyncLibJarsForDebug
:CordovaLib:mergeDebugJniLibFolders
:CordovaLib:transformNative_libsWithMergeJniLibsForDebug
:CordovaLib:transformNative_libsWithSyncJniLibsForDebug
:CordovaLib:bundleDebug
:CordovaLib:preReleaseBuild UP-TO-DATE
:CordovaLib:checkReleaseManifest
:CordovaLib:prepareReleaseDependencies
:CordovaLib:compileReleaseAidl
:CordovaLib:compileReleaseNdk UP-TO-DATE
:CordovaLib:copyReleaseLint UP-TO-DATE
:CordovaLib:mergeReleaseShaders
:CordovaLib:compileReleaseShaders
:CordovaLib:generateReleaseAssets
:CordovaLib:mergeReleaseAssets
:CordovaLib:mergeReleaseProguardFiles
:CordovaLib:packageReleaseRenderscript UP-TO-DATE
:CordovaLib:compileReleaseRenderscript
:CordovaLib:generateReleaseResValues
:CordovaLib:generateReleaseResources
:CordovaLib:packageReleaseResources
:CordovaLib:processReleaseManifest
:CordovaLib:generateReleaseBuildConfig
:CordovaLib:processReleaseResources
:CordovaLib:generateReleaseSources
:CordovaLib:incrementalReleaseJavaCompilationSafeguard
:CordovaLib:compileReleaseJavaWithJavac
:CordovaLib:compileReleaseJavaWithJavac - is not incremental (e.g. outputs have changed, no previous execution, etc.).
Note: Some input files use or override a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
:CordovaLib:processReleaseJavaRes UP-TO-DATE
:CordovaLib:transformResourcesWithMergeJavaResForRelease
:CordovaLib:transformClassesAndResourcesWithSyncLibJarsForRelease
:CordovaLib:mergeReleaseJniLibFolders
:CordovaLib:transformNative_libsWithMergeJniLibsForRelease
:CordovaLib:transformNative_libsWithSyncJniLibsForRelease
:CordovaLib:bundleRelease
:prepareComAndroidSupportAnimatedVectorDrawable2602Library
:prepareComAndroidSupportAppcompatV72602Library
:prepareComAndroidSupportCardviewV72510Library
:prepareComAndroidSupportDesign2602Library
:prepareComAndroidSupportRecyclerviewV72602Library
:prepareComAndroidSupportSupportCompat2602Library
:prepareComAndroidSupportSupportCoreUi2602Library
:prepareComAndroidSupportSupportCoreUtils2602Library
:prepareComAndroidSupportSupportFragment2602Library
:prepareComAndroidSupportSupportMediaCompat2602Library
:prepareComAndroidSupportSupportV42602Library
:prepareComAndroidSupportSupportVectorDrawable2602Library
:prepareComAndroidSupportTransition2602Library
:prepareComGoogleAndroidGmsPlayServicesBase1122Library
:prepareComGoogleAndroidGmsPlayServicesBasement1122Library
:prepareComGoogleAndroidGmsPlayServicesGcm1122Library
:prepareComGoogleAndroidGmsPlayServicesIid1122Library
:prepareComGoogleAndroidGmsPlayServicesLocation1122Library
:prepareComGoogleAndroidGmsPlayServicesTasks1122Library
:prepareComIntercomAndroidComposer131Library
:prepareComIntercomAndroidComposerGallery131Library
:prepareComUrbanairshipAndroidUrbanairshipSdk825Library
:prepareIoIntercomAndroidIntercomSdkBase404Library
:prepareOrgApacheCordovaCordovaLib623DebugLibrary
:prepareDebugDependencies
:compileDebugAidl
:compileDebugRenderscript
:generateDebugBuildConfig
:generateDebugResValues
:generateDebugResources
:mergeDebugResources
:processDebugManifest
/project/AndroidManifest.xml:66:9-76:20 Warning:
	activity#com.urbanairship.cordova.CustomLandingPageActivity was tagged at AndroidManifest.xml:66 to replace another declaration but no other declaration present
:processDebugResources/project/build/intermediates/res/merged/debug/values-v26/values-v26.xml:15:21-54: AAPT: No resource found that matches the given name: attr 'android:keyboardNavigationCluster'.
    
/project/build/intermediates/res/merged/debug/values-v26/values-v26.xml:18:21-54: AAPT: No resource found that matches the given name: attr 'android:keyboardNavigationCluster'.
    
/project/build/intermediates/res/merged/debug/values-v26/values-v26.xml:15: error: Error: No resource found that matches the given name: attr 'android:keyboardNavigationCluster'.

/project/build/intermediates/res/merged/debug/values-v26/values-v26.xml:18: error: Error: No resource found that matches the given name: attr 'android:keyboardNavigationCluster'.


 FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':processDebugResources'.
> com.android.ide.common.process.ProcessException: Failed to execute aapt

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output.

BUILD FAILED

Total time: 10.645 secs
Error: /project/gradlew: Command failed with exit code 1 Error output:
Note: Some input files use or override a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
Note: Some input files use or override a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
/project/build/intermediates/res/merged/debug/values-v26/values-v26.xml:15:21-54: AAPT: No resource found that matches the given name: attr 'android:keyboardNavigationCluster'.
    
/project/build/intermediates/res/merged/debug/values-v26/values-v26.xml:18:21-54: AAPT: No resource found that matches the given name: attr 'android:keyboardNavigationCluster'.
    
/project/build/intermediates/res/merged/debug/values-v26/values-v26.xml:15: error: Error: No resource found that matches the given name: attr 'android:keyboardNavigationCluster'.

/project/build/intermediates/res/merged/debug/values-v26/values-v26.xml:18: error: Error: No resource found that matches the given name: attr 'android:keyboardNavigationCluster'.



FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':processDebugResources'.
> com.android.ide.common.process.ProcessException: Failed to execute aapt

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output.
    at ChildProcess.whenDone (/project/cordova/node_modules/cordova-common/src/superspawn.js:169:23)
    at emitTwo (events.js:106:13)
    at ChildProcess.emit (events.js:191:7)
    at maybeClose (internal/child_process.js:877:16)
    at Process.ChildProcess._handle.onexit (internal/child_process.js:226:5)
```

**Comment by ronocod** [comment]
This is the same issue as #204, you can find the fix in this comment: https://github.com/intercom/intercom-cordova/issues/204#issuecomment-329178158

**Comment by jskrepnek** [comment]
Thanks @ronocod.  Since we use Phonegap Build, we're a bit removed from those kinds of solutions.       The build was run with cordova-android 6.2.3, which does support android sdk tool 26.0.1, and the targetSdkVersion was set to 26.  Will see if I can get this going locally, but would like to hear from other build users.

**Comment by jlf123** [comment]
I'm having the same issue, ay update on this?

**Comment by jskrepnek** [comment]
No luck yet.  Probably need to wait for Build to support [cordova-android 6.3.0](https://github.com/apache/cordova-android/blob/master/RELEASENOTES.md#release-notes-for-cordova-android).  Not sure when that will be.

**Comment by ronocod** [comment]
Closing this, the Cordova version that Build supports is out of our control so there's nothing we can do.

**Comment by jskrepnek** [comment]
Hey @ronocod, do we *need* to target API 26 to use version 4+ of the plugin?  There seems to be confusion between having support for API 26 and support for build tools (sometimes called sdk tools) v26.  Support for build tools v26 was included in [cordova android 6.2.3](https://cordova.apache.org/announcements/2017/05/04/android-release.html) (phonegap 7.0.1).

It seems like we should be able to use the new version of the plugin without having to target API 26.

**Comment by ronocod** [comment]
We only officially support
* Targetting API 26 (`compileSdkVersion 26` in `build.gradle`)
* Using version 26.0.0 or higher `buildToolsVersion`
* Using Support Libraries with version 26.0.0 or higher

The SDK may work if your build setup doesn't match that, but we can't make any guarantees. We've seen lots of customer issues caused by build setups that used lower versions of the above dependencies.
